common/persistence/sql/sqlplugin/mysql/queue.go

```diff
@@ -32,7 +32,9 @@ const (
 	templateEnqueueMessageQuery            = `INSERT INTO queue (queue_type, message_id, message_payload) VALUES(:queue_type, :message_id, :message_payload)`
 	templateGetLastMessageIDQuery          = `SELECT message_id FROM queue WHERE message_id >= (SELECT message_id FROM queue WHERE queue_type=? ORDER BY message_id DESC LIMIT 1) FOR UPDATE`
 	templateGetMessagesQuery               = `SELECT message_id, message_payload FROM queue WHERE queue_type = ? and message_id > ? LIMIT ?`
-	templateDeleteMessagesQuery            = `DELETE FROM queue WHERE queue_type = ? and message_id < ?`
+	templateGetMessagesBeforeQuery         = `SELECT message_id, message_payload FROM queue WHERE queue_type = ? and message_id <= ?`
```

Comment:
do you need the messages to be ordered by message_id ? If yes, you need to add ORDER BY

Reply:
Yes. The ordering is needed. I checked Cassandra do ascending order by default. Do you think I need to add this to cassandra query?

Reply:
This will return all the messages, don't you need batching and/or LIMIT ?

Reply:
One assumption is the DLQ won't have a large amount of messages. As this is not a valid assumption for a queue. I will add both boundaries in the query and limit the size